src/App.tsx

```diff
@@ -5,7 +5,7 @@ const App = () => {
     <>
       <div className="App">
         <h1>FoundersLab</h1>
-        <Link to="/new-user">Get Started</Link>
+        <Link to="/login">Get Started</Link>
       </div>
     </>
   );
```

Comment:
It'll be better to use Routes declaration inside `App.tsx` file. 
Assume this as an file where we will define all our page routers.

The `main.tsx` file should be like this - 

``` jsx
import { BrowserRouter as Router } from 'react-router-dom';


ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
      <Router>
        <App />
      </Router>
  </React.StrictMode>
);
```

And `App.tsx` file should look like this -

``` jsx
const App = () => {
  return (
    <Suspense fallback={<Loading />}>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route
          path="/dashboard"
          element={
            <PrivateRoute>
              <Dashboard />
            </PrivateRoute>
          }
        />
        <Route
          path="/new-user"
          element={<NewUser />}
        />
        <Route path="/login" element={<Login />} />
      </Routes>
    </Suspense>
  );
}
```

Also, the routes which are **protected** needs to be wrapped with a `<PrivateRoute />` component in which we check whether the user is logged in.

PrivateRoute will be something like this -

``` jsx
const PrivateRoute: FC<ComponentProps> = ({ children }) => {
  
  // We will be getting the user/token or something to check whether the user is logged in or not
  const user = getUserExampleFunction();

  if (user === undefined) {
    // TODO - Add Loading spinner
    return <p>Loading pvt route...</p>;
  }

  // If user is there then we return the children
  if (user) {
    return children;
  }

  // If user is not logged in, we move the user to '/'
  return <Navigate to="/" />;
};
```

Can read more about all these from the react-router-dom v6 docs.

Also, use lazy-loading while importing these page routes so that we can use code-splitting.